ompi/mca/op/aarch64/op_aarch64_functions.c

```diff
@@ -140,20 +142,18 @@ _Generic((*(out)), \
                              struct ompi_datatype_t **dtype,                      \
                              struct ompi_op_base_module_1_0_0_t *module)          \
     {                                                                             \
-        int types_per_step = svcnt(*((type##type_size##_t *) _in));               \
-        size_t idx = 0, left_over = *count;                                       \
+        const int types_per_step = svcnt(*((type##type_size##_t *) _in));         \
+        const int cnt = *count;                                                   \
         type##type_size##_t *in = (type##type_size##_t *) _in,                    \
                             *out = (type##type_size##_t *) _out;                  \
         OP_CONCAT(OMPI_OP_TYPE_PREPEND, type##type_size##_t) vsrc, vdst;          \
-        svbool_t pred = svwhilelt_b##type_size(idx, left_over);                   \
-        do {                                                                      \
+        for (int idx=0; idx < *count; idx += types_per_step) {                    \
```

Comment:
Should this be using `cnt` instead?
```suggestion
        for (int idx=0; idx < cnt; idx += types_per_step) {                    \
```
I don't see it being used anywhere else

Reply:
I missed that. Still, the other loop touched here uses `cnt` as upper bound: https://github.com/open-mpi/ompi/pull/12683/files#diff-8ce11c0a08297c59d91f9d3a8c67b119707fea2a2dbe0cdab217b5449f218e48R317

Reply:
It should be cnt indeed. Will update the PR tomorrow.

Reply:
Makes sense to have `cnt` there.